app/models/mail_notification.rb

```diff
@@ -29,6 +29,10 @@ def self.[](name)
     self.find_by_name(name)
   end
 
+  def puppet_error_state?
```

Comment:
How about subclassing MailNotification? Name/type checking is kind of a giveaway you're dealing with a different object.